stonesoup/updater/kalman.py

```diff
@@ -628,3 +653,136 @@ def update(self, hypothesis, **kwargs):
+
+    def _posterior_mean(self, predicted_state, kalman_gain, measurement, measurement_prediction):
+        """Compute the posterior mean, :math:`s_{k|k} = s_{k|k-1} + K_s (z - H_s s_{k|k-1} -
+        H_p p_{k|k-1})`, :math:`p_{k|k} = p_{k|k-1}.
+
+        Parameters
+        ----------
+        predicted_state : :class:`State`, :class:`Prediction`
+            The predicted state
+        kalman_gain : numpy.ndarray
+            The reduced form of the Kalman gain, :math:`K_s`
+         measurement : :class:`Detection`
+            The measurement
+        measurement_prediction : :class:`MeasurementPrediction`
+            Predicted measurement
+
+        Returns
+        -------
+        : :class:`StateVector`
+            The posterior mean estimate
+        """
+        post_mean = predicted_state.state_vector.copy()
+        post_mean[np.ix_(~self.consider)] += \
+            kalman_gain @ (measurement.state_vector - measurement_prediction.state_vector)
+        return post_mean.view(StateVector)
+
+    def _posterior_covariance(self, hypothesis):
+        """
+        Return the posterior covariance for a given hypothesis
+
+        Parameters
+        ----------
+        hypothesis: :class:`~.Hypothesis`
+            A hypothesised association between state prediction and measurement. It returns the
+            measurement prediction which in turn contains the measurement cross covariance,
+            :math:`P_{k|k-1} H_k^T and the innovation covariance,
+            :math:`S = H_k P_{k|k-1} H_k^T + R`
+
+        Returns
+        -------
+        : :class:`~.CovarianceMatrix`
+            The posterior covariance matrix rendered via the Schmidt-Kalman update process.
+        : numpy.ndarray
+            The reduced form of the Kalman gain,
+            :math:`K_s = (P_{ss,k|k-1} H_{s,k}^T + P_{sp,k|k-1} H_{p,k}^T) S^{-1}`
+
+        """
+        # Intermediate matrices P_p and H.
+        pp = hypothesis.prediction.covar[np.tile(self.consider, (len(self.consider), 1))]
+        pp = pp.reshape((len(self.consider), np.sum(self.consider)))
+        hh = self._measurement_matrix(predicted_state=hypothesis.prediction)
+
+        # First get the Kalman gain
+        mcc = hypothesis.measurement_prediction.cross_covar
+        kalman_gain = mcc[np.ix_(~self.consider)] @ \
+            np.linalg.inv(hypothesis.measurement_prediction.covar)
+
+        # Then assemble the quadrants of the posterior covariance (easier to think of them as
+        # quadrants even though they're actually submatrices who may appear in somewhat different
+        # places.)
+        post_cov = hypothesis.prediction.covar.copy()
+        post_cov[np.ix_(~self.consider, ~self.consider)] -= \
+            kalman_gain @ hypothesis.measurement_prediction.covar @ kalman_gain.T
+        post_cov[np.ix_(~self.consider, self.consider)] -= kalman_gain @ hh @ pp
+        post_cov[np.ix_(self.consider, ~self.consider)] -= pp.T @ hh.T @ kalman_gain.T
```

Comment:
Saves a bit more time  `(AB)ᵀ=BᵀAᵀ`

```suggestion
        khp = kalman_gain @ hh @ pp
        post_cov[np.ix_(~self.consider, self.consider)] -= khp
        post_cov[np.ix_(self.consider, ~self.consider)] -= khp.T
```